src/main/kotlin/com/terraformation/backend/search/NestedQueryBuilder.kt

```diff
@@ -270,19 +270,19 @@ import org.jooq.impl.DSL
  * on bag numbers (returning a result with a single bag) or treat it as a filter on accessions
  * (returning a result with two bags).
  *
- * Our product decision is to do the latter. Search criteria control which top-level results are
- * returned, but each result includes the full set of values for all its fields. When you search for
- * bag "A", what you're really telling the system is that you want all _accessions_ that have a bag
- * called "A". And for each accession that matches the search criteria, you get back the full list
- * of bags.
+ * Our product decision was originally to do the latter. Search criteria would control which
```

Comment:
This would be a more substantial edit, but I think it might be better to rework this section to say that we support both models and describe how. Someone reading this a year from now won't care about outdated product decisions; they'll just want to know how the code currently works.

Or maybe there's no need to document this as two different modes of operation at all any more. Now that you can filter the results on any prefix, the root prefix isn't anything special. So we can just document how the per-prefix filtering works, and maybe add a brief note that filtering on the root prefix will determine the list of top-level results.

Reply:
Will rework 👍